pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -90,6 +90,17 @@ def __verify_train_loop_configuration(self, model: "pl.LightningModule") -> None
                 "(rather, they are called on every optimization step)."
             )
 
+        # Allow max_epochs or max_steps to be zero, since this will be handled by fit_loop.done
+        if trainer.max_epochs and (trainer.max_epochs < -1 or not isinstance(trainer.max_epochs, int)):
```

Comment:
Neat !

Reply:
is the isinstance check for int needed? given the type is Optional[int] ?

Reply:
I think yes because of https://github.com/PyTorchLightning/pytorch-lightning/pull/8877#discussion_r697930369

Reply:
+1 for removing the strict type checks - we already avoid them in other parts of the codebase

checking this is responsibility of the IDEs/mypy as Python users.